PROTO_tests/tests/indexing_test.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+NUM_TYPES = [ak.int64, ak.uint64, ak.float64, ak.bool, ak.bigint]
+BIGINT_AND_UINT_TYPES = [ak.bigint, ak.uint64]
+
+
+def key_arrays(size):
+    ikeys = ak.arange(size)
+    ukeys = ak.arange(size, dtype=ak.uint64)
+    return ikeys, ukeys
+
+
+def value_array(num_types, size):
+    array_dict = {
+        ak.int64: (i := ak.randint(0, size, size)),
+        ak.uint64: (u := ak.cast(i, ak.uint64)),
+        ak.float64: ak.array(np.random.randn(size)),
+        ak.bool: (i % 2) == 0,
+        ak.bigint: ak.cast(u, ak.bigint),
+        ak.str_: ak.cast(i, str),
+    }
+    return array_dict[num_types]
+
+
+class TestIndexing:
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("num_types", NUM_TYPES)
+    def test_pdarray_uint_indexing(self, prob_size, num_types):
+        # for every pda in array_dict test indexing with uint array and uint scalar
+        ikeys, ukeys = key_arrays(prob_size)
+        pda = value_array(num_types, prob_size)
+        assert pda[np.uint(2)] == pda[2]
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_strings_uint_indexing(self, prob_size):
+        # test Strings array indexing with uint array and uint scalar
+        ikeys, ukeys = key_arrays(prob_size)
+        pda = value_array(ak.str_, prob_size)
+        assert pda[np.uint(2)] == pda[2]
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
```

Comment:
there's no reason these need to be separate tests, we could either create a `NUM_AND_STR_TYPES` or just do 
```python
@pytest.mark.parametrize("prob_size", pytest.prob_size)
@pytest.mark.parametrize("num_types", NUM_TYPES + [ak.str_])
def test_pdarray_uint_indexing(self, prob_size, num_types):
    # for every pda in array_dict test indexing with uint array and uint scalar
    ikeys, ukeys = key_arrays(prob_size)
    pda = value_array(num_types, prob_size)
    assert pda[np.uint(2)] == pda[2]
    assert pda[ukeys].to_list() == pda[ikeys].to_list()
```

Reply:
Updated